### PR TITLE
feat: raspixud

### PR DESCRIPTION
**Pull request by kilrau**
Dependency: https://github.com/ExchangeUnion/xud-docker/issues/245
Dependency: https://github.com/ExchangeUnion/xud-docker/issues/264

### CODE REVIEW

File: Market Maker Guide.md [hunk 1]
```diff
@@ -1,40 +1,73 @@
...
+This guide is written for individuals and entities looking to run xud and become a market maker on OpenDEX. Occasional traders can use this guide to run xud to trade on OpenDEX.
 
-# Supported Networks
+# Prerequisites
 
-## Simnet
-Private chains which are maintained by Exchange Union. We’ll automatically open channels to you and push over some coins, you’ll be trading against our bots and anyone else running simnet. It’s the perfect play money playground to see how things work and play around with `xucli` commands. It’s easy: run the launch script, wait for about 15 minutes and you are ready to go. **You want to start with this!**
+## Two Modes
+1. **Light Setup**, using [Neutrino](https://github.com/lightninglabs/neutrino) and [Infura](https://infura.io/). This keeps the setup light, but is not fully trustless.
+2. **Full Setup**, using [bitcoind](https://github.com/bitcoin/bitcoin/), [litecoind](https://github.com/litecoin-project/litecoin) and [geth](https://github.com/ethereum/go-ethereum). Requires more resources and a SSD, but keeps the setup trustless.
```
Comment (ghost) [inline Market Maker Guide.md]:
```suggestion
2. **Full Setup**, using [bitcoind](https://github.com/bitcoin/bitcoin/), [litecoind](https://github.com/litecoin-project/litecoin) and [geth](https://github.com/ethereum/go-ethereum). Requires more resources and an SSD, but keeps the setup trustless.
```

File: Market Maker Guide.md [hunk 2]
```diff
@@ -1,40 +1,73 @@
...
+* Docker >= 18.09. Check with `docker --version`. If you do not have docker installed yet, follow the official [install instructions](https://docs.docker.com/install/). Also make sure that the current user can run docker (without adding `sudo`). Test with `docker run hello-world`. If this fails, [follow these instructions](https://docs.docker.com/install/linux/linux-postinstall/).
 
-# Basic Setup
+## Hardware
+* Since market makers should be online 24/7 and we are ushering in a post-cloud era, we recommend setting up a power-efficient linux box connected to your router. We recommend building your own xud-in-a-box with a Raspberry Pi following the [RaspiXUD guide](RaspiXUD.md).
+* If you are using a different device (take the [RaspiXUD guide](RaspiXUD.md) as reference) or a cloud VPS: We support `x64` (also called `amd64`) and `arm64` (also called `aarch64`) devices, which should cover most of the devices. We recommend >=16GB RAM and a SSD for the full setup, 4GB RAM and HDD for the light setup.
```
Comment (ghost) [inline Market Maker Guide.md]:
```suggestion
* If you are using a different device (take the [RaspiXUD guide](RaspiXUD.md) as reference) or a cloud VPS: We support `x64` (also called `amd64`) and `arm64` (also called `aarch64`) devices, which should cover most of the devices. We recommend >=16GB RAM and an SSD for the full setup, 4GB RAM and HDD for the light setup.
```

File: RaspiXUD.md [hunk 3]
```diff
@@ -0,0 +1,161 @@
+This guide is written for market makers & traders to build their own xud-in-a-box with a Raspberry Pi.
+
+![RaspiXUD](images/RaspiXUD.jpeg)
+
+Two options are available:
+1. **Light setup** using [Neutrino](https://github.com/lightninglabs/neutrino) and [Infura](https://infura.io/). This keeps the setup light and cheap (~100 EUR), but is not fully trustless. A Pi3 works fine for this.
+2. **Full setup** using [bitcoind](https://github.com/bitcoin/bitcoin/), [litecoind](https://github.com/litecoin-project/litecoin) and [geth](https://github.com/ethereum/go-ethereum). Requires a SSD, but keeps the setup trustless. You'll need a Pi4 with 4GB of RAM.
```
Comment (ghost) [inline RaspiXUD.md]:
```suggestion
2. **Full setup** using [bitcoind](https://github.com/bitcoin/bitcoin/), [litecoind](https://github.com/litecoin-project/litecoin) and [geth](https://github.com/ethereum/go-ethereum). Requires an SSD, but keeps the setup trustless. You'll need a Pi4 with 4GB of RAM.
```

**Review by ghost — APPROVED** [review]
Did not test it manually with physical hardware, but looks good in general after reading it end-to-end.

File: RaspiXUD.md [hunk 4]
```diff
@@ -0,0 +1,161 @@
+This guide is written for market makers & traders to build their own xud-in-a-box with a Raspberry Pi.
+
+![RaspiXUD](images/RaspiXUD.jpeg)
+
+Two options are available:
+1. **Light setup** using [Neutrino](https://github.com/lightninglabs/neutrino) and [Infura](https://infura.io/). This keeps the setup light and cheap (~100 EUR), but is not fully trustless. A Pi3 works fine for this.
+2. **Full setup** using [bitcoind](https://github.com/bitcoin/bitcoin/), [litecoind](https://github.com/litecoin-project/litecoin) and [geth](https://github.com/ethereum/go-ethereum). Requires an SSD, but keeps the setup trustless. You'll need a Pi4 with 4GB of RAM.
+
+If you are not sure, we recommend to start with the light setup. If you opt for the Pi4 4GB, you can swith to the full setup any time.
```
Comment (dopetard) [inline RaspiXUD.md]: ~~swith~~ switch

**Review by dopetard — DISMISSED** [review]
Haven't tested it yet but without ethernet cable, there is additional wifi config is necessary after flashing sd card.  Might be worth mentioning. Looks good.

File: RaspiXUD.md (hunk 4 above)
Comment (kilrau) [reply RaspiXUD.md]: Fixed.

**Comment by kilrau** [comment]
> Haven't tested it yet but without ethernet cable, there is additional wifi config is necessary after flashing sd card. Might be worth mentioning. Looks good.

Added.

File: RaspiXUD.md [hunk 5]
```diff
@@ -0,0 +1,183 @@
...
+ubuntu@ubuntu:~$ sudo su - xud
+xud@ubuntu:~$ docker run hello-world
+Hello from Docker!
+This message shows that your installation appears to be working correctly.
+```
+10. Looking good! Optionally, add an alias to enter your xud environment by simply typing "xud":
+```bash
+xud@ubuntu:~$ sudo nano ~/.bash_aliases
+# add the line
+alias xud='bash ~/xud.sh'
```
Comment (reliveyy) [inline RaspiXUD.md:99]: Is this `xud.sh` strange to the user coming directly to this doc and know nothing about the script file is inside xud-docker?

File: RaspiXUD.md [hunk 6]
```diff
@@ -0,0 +1,183 @@
...
+# CTRL+S, CTRL+X. Then run
+xud@ubuntu:~$ source ~/.bashrc
+```
+11. Connect an ext4-formatted USB stick to your Pi and set it up. It is very important to do this (given you do not want to loose money)!
```
Comment (reliveyy) [inline RaspiXUD.md]: Does the filesystem have to be ext4 only? The USB stick could be FAT/exFAT format for some users becuase the want to access the stick content on their Windows or macOS.

File: RaspiXUD.md [hunk 7]
```diff
@@ -0,0 +1,183 @@
...
+# check if mounting worked
+xud@ubuntu:~$ df -h
+# make sure xud can use it
+xud@ubuntu:~$ sudo chown xud:xud /media/USB
+```
+12. Light setup - **DONE!** Continue [here](Market%20Maker%20Guide.md#the-setup).
+
+## Pi Full Setup
+
+13. Connect the ext4-formatted SSD (one large partition) to your Pi4 and set it up:
```
Comment (reliveyy) [inline RaspiXUD.md]: Same as before. E.g. I'd like to use btrfs for my HDD and SSD. This is a filesystem which gonna be widely used in future Linux to replace ext4.

File: Market Maker Guide.md [hunk 8]
```diff
@@ -78,30 +127,74 @@ After this, the setup pulls docker containers, starts syncing chains and opens
...
+│ xud       │ Waiting for sync                               │
+└───────────┴────────────────────────────────────────────────┘
+```
+The full setup starts syncing fast and gets slower towards the end. Bitcoind/Litecoind should finish syncing within 12h, whereas geth will need about 3 full days. The light setup should be ready within 2-3h (there is no working light mode for LTC yet). A Pi4 needs about twice that long.
+```
+mainnet > status
+┌───────────┬────────────────────────────────────────────────┐
+│ SERVICE   │ STATUS                                         │
+├───────────┼────────────────────────────────────────────────┤
+│ bitcoind  │ Ready (Connected to neutrino)                  │
```
Comment (reliveyy) [inline Market Maker Guide.md]: It's capitalized Neutrino.

File: Market Maker Guide.md [hunk 9]
```diff
@@ -78,30 +127,74 @@ After this, the setup pulls docker containers, starts syncing chains and opens
...
+├───────────┼────────────────────────────────────────────────┤
+│ litecoind │ Ready (Connected to external)                  │
+├───────────┼────────────────────────────────────────────────┤
+│ geth      │ Ready (Connected to infura)                    │
```
Comment (reliveyy) [inline Market Maker Guide.md]: It's capitalized Infura.

File: RaspiXUD.md (hunk 6 above)
Comment (kilrau) [reply RaspiXUD.md]: True, removed that.

File: RaspiXUD.md (hunk 7 above)
Comment (kilrau) [reply RaspiXUD.md]: Fixed.

File: RaspiXUD.md (hunk 5 above)
Comment (kilrau) [reply RaspiXUD.md:99]: I think it's fine.

File: Market Maker Guide.md (hunk 8 above)
Comment (kilrau) [reply Market Maker Guide.md]: Fixed.

File: RaspiXUD.md [hunk 10]
```diff
@@ -0,0 +1,183 @@
+This guide is written for market makers & traders to turn a Raspberry Pi into an always-on xud node.
+
+![RaspiXUD setup, plants are optional](images/RaspiXUD.jpeg)
+
+Two options are available:
+1. **Light setup** using [Neutrino](https://github.com/lightninglabs/neutrino) and [Infura](https://infura.io/). This keeps the setup light-weight & cheap, but requires to trust entities like [Infura](https://infura.io/).
+2. **Full setup** using [bitcoind](https://github.com/bitcoin/bitcoin/), [litecoind](https://github.com/litecoin-project/litecoin) and [geth](https://github.com/ethereum/go-ethereum). Requires more resources and an SSD, but keeps the setup trustless.
+
+If you are not sure, we recommend to start with the light setup. If you opt for the Pi4 4GB, you can switch to the full setup any time.
```
Comment (ghost) [inline RaspiXUD.md]:
```suggestion
If you are not sure, we recommend to start with the light setup. If you opt for the Pi4 4GB, you can switch to the full setup at any time.
```

**Review by ghost — APPROVED** [review]

**Pull request merged**